src/irr/asset/CGLTFLoader.cpp

```diff
@@ -317,16 +323,34 @@ namespace irr
 					if (glTFprimitive.indices.has_value())
 					{
 						auto& glTFIndexAccessor = glTFprimitive.accessors["INDEX"];
+						handleAccessor(glTFIndexAccessor);
 
-						// TODO
+						switch (glTFIndexAccessor.componentType.value())
+						{
+							case SGLTFAccessor::SCT_UNSIGNED_SHORT:
+							{
+								cpuMeshBuffer->setIndexType(EIT_16BIT);
+							} break;
+
+							case SGLTFAccessor::SCT_UNSIGNED_INT:
+							{
+								cpuMeshBuffer->setIndexType(EIT_32BIT);
+							} break;
```

Comment:
remember to validate that the stride matches the size (error out or repack the buffer otherwise)